packages/adapters-library/src/adapters/stargate/products/farm/stargateFarmAdapter.ts

```diff
@@ -0,0 +1,454 @@
+    return await this.getMovements({
+      ...input,
+      filterType: 'deposit',
+    })
+  }
+
+  private async getMovements({
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+    userAddress,
+    filterType,
+  }: GetEventsInput & {
+    filterType: 'deposit' | 'withdrawal'
+  }): Promise<MovementsByBlock[]> {
+    const protocolToken =
+      await this.getProtocolTokenByAddress(protocolTokenAddress)
+
+    const lpStakingContract =
+      protocolToken.lpStakingType === 'LpStaking'
+        ? LpStaking__factory.connect(
+            protocolToken.lpStakingAddress,
+            this.provider,
+          )
+        : LpStakingTime__factory.connect(
+            protocolToken.lpStakingAddress,
+            this.provider,
+          )
+
+    const filter =
+      filterType === 'deposit'
+        ? lpStakingContract.filters.Deposit(
+            userAddress,
+            protocolToken.poolIndex,
+            undefined,
+          )
+        : lpStakingContract.filters.Withdraw(
+            userAddress,
+            protocolToken.poolIndex,
+            undefined,
+          )
+
+    const events = await lpStakingContract.queryFilter(
+      filter,
+      fromBlock,
+      toBlock,
+    )
+
+    return events.map((event) => {
+      const { amount } = event.args!
+
+      return {
+        protocolToken: {
+          address: protocolToken.address,
+          name: protocolToken.name,
+          symbol: protocolToken.symbol,
+          decimals: protocolToken.decimals,
+        },
+        blockNumber: event.blockNumber,
+        transactionHash: event.transactionHash,
+        tokens: [
+          {
+            address: protocolToken.address,
+            name: protocolToken.name,
+            symbol: protocolToken.symbol,
+            decimals: protocolToken.decimals,
+            type: TokenType.Underlying,
+            blockNumber: event.blockNumber,
+            balanceRaw: amount,
+          },
+        ],
+      }
+    })
+  }
+
+  /**
+   * There are no specific events for rewards, which occur as simple transfers whenever a Deposit or Withdrawal is actioned
+   * For this reason, we need to fetch all Deposit and Withdrawal events and then filter out the reward transfers
+   */
+  async getRewardWithdrawals({
+    userAddress,
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+  }: GetEventsInput): Promise<MovementsByBlock[]> {
+    const protocolToken =
+      await this.getProtocolTokenByAddress(protocolTokenAddress)
+
+    const lpStakingContract =
+      protocolToken.lpStakingType === 'LpStaking'
+        ? LpStaking__factory.connect(
+            protocolToken.lpStakingAddress,
+            this.provider,
+          )
+        : LpStakingTime__factory.connect(
+            protocolToken.lpStakingAddress,
+            this.provider,
+          )
+
+    const depositFilter = lpStakingContract.filters.Deposit(
+      userAddress,
+      protocolToken.poolIndex,
+      undefined,
+    )
```

Comment:
Just to confirm are you saying rewards can be "claimed" even with deposits?

Reply:
Yes, rewards get automatically claimed whenever user deposits or withdraws. They don't seem to have a method to claim them directly.
![image](https://github.com/user-attachments/assets/1c07783b-0694-4a6a-9c2a-5de2fca00a02)